build.gradle

```diff
@@ -298,20 +255,38 @@ subprojects {
         }
     }
 
-    publishing {
-        publications {
-            if (![project(":digdag-docs")].contains(project)) {
+    // digdag-docs is out of target on publish
+    if (![project(":digdag-docs")].contains(project)) {
+        publishing {
+            publications {
                 mavenJava(MavenPublication) {
                     from components.java
                     afterEvaluate {
                         artifact testsJar
                         artifact sourcesJar
                         artifact javadocJar
-
                         pom.withXml(pomXmlModifier)
                     }
                 }
             }
+            repositories {
+                maven {  // publishMavenJavaPublicationToMavenCentralRepository
+                    name = "mavenCentral"
+                    if (project.ext.isSnapshotRelease) {
+                        url "https://oss.sonatype.org/content/repositories/snapshots"
+                    } else {
+                        url "https://oss.sonatype.org/service/local/staging/deploy/maven2"
+                    }
+                    credentials {
+                        username = project.hasProperty("ossrhUsername") ? ossrhUsername : ""
+                        password = project.hasProperty("ossrhPassword") ? ossrhPassword : ""
+                    }
+                }
+            }
+        }
+        signing {
```

Comment:
If you want `gradle publishToMavenLocal` to work without `signing.gnupg` properties, this would be a workaround
```
    if (project.hasProperty('signing.gnupg.keyName')) {
        signing {
            :     
```

Reply:
Good point! We need to publish in local and it is useful fix.

Reply:
I fixed it.